### PR TITLE
Bug 1337879, added manual upgrade steps to get the latest templates f…

### PR DESCRIPTION
**Pull request by ahardin-rh**
…rom _openshift-ansible-roles_

https://bugzilla.redhat.com/show_bug.cgi?id=1337879

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@sdodson PTAL. Thanks!

File: install_config/upgrading/manual_upgrades.adoc [hunk 1]
```diff
@@ -748,6 +748,37 @@ ifdef::openshift-origin[]
...
+Ansible. you need to get the latest .json files on the system where you are
+running `oc`, which can run anywhere that has access to the master.
+
+Install *atomic-openshift-utils* and its dependencies to install the new content
+into
+*_/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/_*.:
+
+====
+----
+$ oc create -n openshift -f  /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/image-streams/image-streams-rhel7.json
```
Comment (sdodson) [inline install_config/upgrading/manual_upgrades.adoc]: Unfortunately we don't have a good method to create anything that doesn't exist and update things that do exist so we have to create, then replace. Like this 

```
$ oc create -n openshift -f  /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/image-streams/image-streams-rhel7.json
$ oc replace -n openshift -f  /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/image-streams/image-streams-rhel7.json
```

Comment (sdodson) [inline install_config/upgrading/manual_upgrades.adoc]: Also, we need to update templates too.
```
$ oc create -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/quickstart-templates/
$ oc create -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/db-templates/
$ oc create -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/infrastructure-templates/
$ oc create -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/xpaas-templates/
$ oc create -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/xpaas-streams/
$ oc replace -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/quickstart-templates/
$ oc replace -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/db-templates/
$ oc replace -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/infrastructure-templates/
$ oc replace -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/xpaas-templates/
$ oc replace -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/xpaas-streams/
```

**Comment by ahardin-rh** [comment]
@sdodson Thanks! I made some edits PTAL.

@tnguyen-rh @bfallonf @tpoitras  peer review please. Thanks!

**Comment by sdodson** [comment]
@ahardin-rh looks good. BTW, there will be errors generated for items that already exist, and that's expected, should we note that? It'll look like this
```
[root@ose3-master ~]# oc create -n openshift -f /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/quickstart-templates/
Error from server: error when creating "/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/quickstart-templates/cakephp-mysql.json": templates "cakephp-mysql-example" already exists
Error from server: error when creating "/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/quickstart-templates/cakephp.json": templates "cakephp-example" already exists
Error from server: error when creating "/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/quickstart-templates/dancer-mysql.json": templates "dancer-mysql-example" already exists
Error from server: error when creating "/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/quickstart-templates/dancer.json": templates "dancer-example" already exists
Error from server: error when creating "/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/quickstart-templates/django-postgresql.json": templates "django-psql-example" already exists
```

**Comment by ahardin-rh** [comment]
@sdodson Yes, I think that is helpful information. I added that in. Thanks!

File: install_config/upgrading/manual_upgrades.adoc [hunk 2]
```diff
@@ -760,6 +760,67 @@ ifdef::openshift-origin[]
...
+rpm -ql openshift-ansible-roles | grep examples | grep v1.2
+----
+====
+
+In this example,
+*_/usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.2/image-streams/image-streams-rhel7.json_*
+is the latest file that you want in the latest *openshift-ansible-roles* package.
+
+*_/usr/share/openshift/examples/image-streams/image-streams-rhel7.json_* is not
+owned by a package. It is updated by Ansible. If you are upgrading outside of
```
Comment (bfallonf) [inline install_config/upgrading/manual_upgrades.adoc]: I's suggest:

*file* is not owned by a package, but is updated by Ansible."

**Comment by bfallonf** [comment]
@ahardin-rh Only one, sentence-level comment from me, but also, should the chunk of stuff you added be in a procedure?

**Comment by tpoitras** [comment]
@ahardin-rh @bfallonf +1 for making this a procedure / enumerated list.

**Comment by ahardin-rh** [comment]
@bfallonf @tpoitras Thanks! Fixed and formatted as a procedure. :sunflower:

**Comment by tpoitras** [comment]
@ahardin-rh good stuff! :jack_o_lantern:

**Comment by ahardin-rh** [comment]
Received QE approval

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../install_config/upgrading/manual_upgrades.adoc#install-config-upgrading-manual-upgrades[Upgrading -> Performing Manual Cluster Upgrades]
|Added manual upgrade steps to get the latest templates from `openshift-ansible-roles`.
%

**Comment by adellape** [comment]
@ahardin-rh I'm gonna pick this now (for today's installer errata) because https://github.com/openshift/openshift-docs/pull/2658 wants it. I'm only picking it to the enterprise-3.2 branch, though, so I'm leaving the enterprise-3.1 label here.

**Pull request merged**